core/src/test/scala/unit/kafka/server/ControllerApisTest.scala

```diff
@@ -142,6 +148,199 @@ class ControllerApisTest {
+    request.topics().add(new DeleteTopicState().setName("baz").setTopicId(ZERO_UUID))
+    request.topics().add(new DeleteTopicState().setName("quux").setTopicId(ZERO_UUID))
+    val response = Set(new DeletableTopicResult().setName(null).setTopicId(barId).
+        setErrorCode(Errors.TOPIC_AUTHORIZATION_FAILED.code).
+        setErrorMessage(Errors.TOPIC_AUTHORIZATION_FAILED.message),
+      new DeletableTopicResult().setName("quux").setTopicId(ZERO_UUID).
+        setErrorCode(Errors.TOPIC_AUTHORIZATION_FAILED.code).
+        setErrorMessage(Errors.TOPIC_AUTHORIZATION_FAILED.message),
+      new DeletableTopicResult().setName("baz").setTopicId(ZERO_UUID).
+        setErrorCode(Errors.TOPIC_AUTHORIZATION_FAILED.code).
+        setErrorMessage(Errors.TOPIC_AUTHORIZATION_FAILED.message),
+      new DeletableTopicResult().setName("foo").setTopicId(fooId).
+        setErrorCode(Errors.TOPIC_AUTHORIZATION_FAILED.code).
+        setErrorMessage(Errors.TOPIC_AUTHORIZATION_FAILED.message))
+    assertEquals(response, controllerApis.deleteTopics(request,
+      ApiKeys.DELETE_TOPICS.latestVersion().toInt,
+      false,
+      _ => Set("foo", "baz"),
+      _ => Set.empty).asScala.toSet)
+  }
+
+  @Test
+  def testNotAuthorizedToDeleteWithTopicNotExisting(): Unit = {
+    val barId = Uuid.fromString("VlFu5c51ToiNx64wtwkhQw")
+    val controller = new MockController.Builder().build()
+    val props = new Properties()
+    props.put(KafkaConfig.DeleteTopicEnableProp, "false")
+    val controllerApis = createControllerApis(None, controller)
+    val request = new DeleteTopicsRequestData()
+    request.topics().add(new DeleteTopicState().setName("foo").setTopicId(ZERO_UUID))
+    request.topics().add(new DeleteTopicState().setName("bar").setTopicId(ZERO_UUID))
+    request.topics().add(new DeleteTopicState().setName(null).setTopicId(barId))
+    val expectedResponse = Set(new DeletableTopicResult().setName("foo").
+        setErrorCode(Errors.UNKNOWN_TOPIC_OR_PARTITION.code).
+        setErrorMessage(Errors.UNKNOWN_TOPIC_OR_PARTITION.message),
+      new DeletableTopicResult().setName("bar").
+        setErrorCode(Errors.TOPIC_AUTHORIZATION_FAILED.code).
+        setErrorMessage(Errors.TOPIC_AUTHORIZATION_FAILED.message),
+      new DeletableTopicResult().setName(null).setTopicId(barId).
+        setErrorCode(Errors.UNKNOWN_TOPIC_ID.code).
+        setErrorMessage(Errors.UNKNOWN_TOPIC_ID.message))
+    assertEquals(expectedResponse, controllerApis.deleteTopics(request,
+      ApiKeys.DELETE_TOPICS.latestVersion().toInt,
+      false,
+      _ => Set("foo"),
+      _ => Set.empty).asScala.toSet)
+  }
+
+  @Test
+  def testNotControllerErrorPreventsDeletingTopics(): Unit = {
+    val fooId = Uuid.fromString("vZKYST0pSA2HO5x_6hoO2Q")
+    val barId = Uuid.fromString("VlFu5c51ToiNx64wtwkhQw")
+    val controller = new MockController.Builder().
+      newInitialTopic("foo", fooId).build()
+    controller.setActive(false)
```

Comment:
this controller is mock so disabling active works well for this test. However, I did not observe the check of control activity in production code. Could you share that with me?

Reply:
See `QuorumController.QuorumMetaLogListener` for the callbacks that make the controller active or inactive.